sdks/java/testing/test-utils/build.gradle

```diff
@@ -23,7 +23,7 @@ description = "Apache Beam :: SDKs :: Java :: Test Utils"
 
 dependencies {
   shadow project(path: ":beam-sdks-java-core", configuration: "shadow")
-  shadow library.java.guava
+  compile library.java.guava
```

Comment:
Could you explain what is the difference between "compile" and "shadow"? I thought I understand this but as I read through "Gradle Primier" doc and "BeamModulePlugin.java" file, I start to have doubts. Some questions: 

 - both shadow and compile configurations "shadow" guava, because it is listed in the `DEFAULT_SHADOW_CLOSURE` that is used there, right? So whichever configuration I use it will still get shadowed the same way. What if Guava wasn't listed in the `DEFAULT_SHADOW_CLOSURE`? Would It get shadowed if I use "compile"?
 - you said that using "compile" packages the depencency "within the jar". So what happens with the guava dependency when I use "shadow"?

Reply:
Another part which is important for shading between `compile` and `shadow` is whether the dependency is listed in the `pom.xml` that is uploaded to Maven central for the artifact.
`compile`: Not part of `pom.xml`
`shadow`: Part of `pom.xml`

Reply:
Thanks for the clarification.